gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/jupiter/handlers/api/v4/flow/FlowChain.java

```diff
@@ -81,10 +96,33 @@ public void addHooks(final List<ChainHook> hooks) {
      */
     public Completable execute(ExecutionContext ctx, ExecutionPhase phase) {
         return resolveFlows(ctx)
+            .switchIfEmpty(
+                Flowable.defer(
+                    () -> {
+                        Boolean previousExecutionMatchNoMatch = ctx.getInternalAttribute(CONTEXT_ATTRIBUTES_FLOW_EXECUTION_NO_MATCH);
+                        ctx.setInternalAttribute(CONTEXT_ATTRIBUTES_FLOW_EXECUTION_NO_MATCH, true);
+                        if (
+                            ExecutionPhase.REQUEST == phase &&
+                            interruptIfNoMatch &&
+                            previousExecutionMatchNoMatch != null &&
+                            previousExecutionMatchNoMatch
```

Comment:
I don't understand why we are interrupting only when the previous execution has no match 🤔

Reply:
Same as Julien, I don't understand it.

Reply:
@jgiovaresco Does my explanation given yesterday help you ?

Reply:
I will try to sum up for @leleueri :

The trick is that we have 2 FlowChain: one for the plans and one for the API. We execute plan's flow before api's one. 
We can't throw a 404 if no plan's flow matched because we need to process API's flow. However, if no plan's flow matched neither no api's flow, then we should return a 404.

During the plan flow execution, it can have `CONTEXT_ATTRIBUTES_FLOW_EXECUTION_NO_MATCH` attribute set but if there is no plans flow executed we will set it.
During the API flow execution, if `CONTEXT_ATTRIBUTES_FLOW_EXECUTION_NO_MATCH` is set it means that no plan flow has matched, and therefore if no API flow matches we should return a 404.